packages/components/src/components/SquiggleViewer/ValueWithContextViewer.tsx

```diff
@@ -166,6 +168,13 @@ export const ValueWithContextViewer: FC<Props> = ({
   const isRoot = path.isRoot();
   const taggedName = value.tags.name();
 
+  const exportData = value.tags.exportData();
+
+  const isRootImport =
+    exportData &&
+    exportData.sourceId !== sourceId &&
```

Comment:
AFAICT, this is the only place where you use `sourceId` from `ProjectContext`.

It's not a big deal either way, but it bother me a bit because `ProjectContext` name suggests that it's something project-wide, but `sourceId` is only for a single source.

One way to avoid `sourceId` entirely is to extract it from `rootValue`. `rootValue` should always be set (unless someone tries to render context-less value with `<SquiggleViewer>`, but then other things would fail), and you could use `rootValue.context.sourceId` to access it.